src/server/db_slice.cc

```diff
@@ -488,18 +503,16 @@ tuple<PrimeIterator, ExpireIterator, bool> DbSlice::AddOrFind2(const Context& cn
 
   DbTable& db = *db_arr_[cntx.db_index];
 
-  // If we have some registered onchange callbacks, we must know in advance whether its Find or Add.
-  if (!change_cb_.empty()) {
-    auto res = FindExt(cntx, key);
+  auto res = FindInternal(cntx, key, FindInternalMode::kDontUpdateStats);
```

Comment:
Why do you think we should not update the stats here? maybe not the miss stats but bumpup and hit I believe we should , no?

Reply:
The reason is that [this](https://github.com/dragonflydb/dragonfly/blob/995f3cd8aa9caf616ebe392ca3f795daffeab7ba/src/server/multi_test.cc#L135) test failed, and I do think it's weird for `SET` to apply hits, but it's a matter of decision.
Whatever decision we make, it needs to be consistent, as it turns out that before this change the stats were different based on whether there were callbacks :facepalm:

Reply:
Ok. Maybe we want to update hits/misses only in case we call FindReadOnly.
Cause right now if we call set we dont update the countes and if we call append which invokes FindMutable we will update this stats.
Anyway I believe we do what to update the bumpups stats even if we dont update the hit/miss

Reply:
Makes sense. Done.